onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -200,14 +200,20 @@ class RemoveDuplicateCastTransformer : public GraphTransformer {
 
  private:
   Status ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const override {
-
     auto output_args = graph.GetOutputs();
     InlinedHashSet<const onnxruntime::NodeArg*> graph_outputs;
     graph_outputs.reserve(output_args.size());
     graph_outputs.insert(output_args.begin(), output_args.end());
     const auto graph_outputs_end = graph_outputs.end();
 
-    for (auto& node : graph.Nodes()) {
+    GraphViewer graph_viewer(graph);
+    const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
```

Comment:
Was there an issue that required this change? Creating a GraphViewer costs a topological sort amongst other things. This code isn't performance critical, but curious as to whether we need to pay that cost here.

Reply:
I answered in another comment, I will revert it if removing node is okay for using original way.